libsolidity/formal/SMTEncoder.cpp

```diff
@@ -867,6 +870,93 @@ void SMTEncoder::endVisit(ElementaryTypeNameExpression const& _typeName)
 	m_context.createExpression(_typeName, result.second);
 }
 
+namespace // helpers for SMTEncoder::visitPublicGetter
+{
+
+bool isReturnedFromStructGetter(TypePointer _type)
+{
+	// So far it seems that only Mappings and ordinary Arrays are not returned (from the cases the compiles).
+	auto category = _type->category();
+	if (category == Type::Category::Mapping)
+		return false;
+	if (category == Type::Category::Array)
+		return static_cast<ArrayType const*>(_type)->isByteArray();
```

Comment:
```suggestion
		return dynamic_cast<ArrayType const&>(_type).isByteArray();
```
`static_cast` _should_ be fine, but we favor `dynamic_cast` because of paranoia about crashes, so I think we should stick to it.